src/Chapter4.hs

```diff
@@ -490,6 +507,19 @@ Implement the 'Applicative' instance for our 'List' type.
 -}
 
 
+instance Applicative List where
+    pure :: a -> List a
+    pure x = Cons x Empty
+
+    (<*>) :: List (a -> b) -> List a -> List b
+    (Cons x xs) <*> yss@(Cons _ _) = 
+        fmap x yss `append` (xs <*> yss)
+      where
+        append :: List a -> List a -> List a
+        append Empty as = as
+        append (Cons a as) bs = Cons a (as `append` bs)
```

Comment:
You can move `append` from here to the top-level, so you will be able to reuse it in the `Monad` instance for `List` :recycle: